src/server/services/validation-rules.js

```diff
@@ -6,8 +6,35 @@ const recordValueFormatters = {
   trimWhitespace: (val) => val.trim(),
   removeCommas: (val) => val.replace(/,/g, ''),
   removeSepDashes: (val) => val.replace(/^-/, '').replace(/;\s*-/g, ';'),
-  toLowerCase: (val) => val.toLowerCase(),
-  childCareSpacing: (val) => (val === 'family or childcare' ? 'family or child care' : val)
+  toLowerCase: (val) => val.toLowerCase()
+}
+
+/*
+Structured data recording all the immediate corrections we want to apply to dropdowns.
+There are 2 types of corrections we can apply:
+1) The value in the current worksheet is incorrect.
+2) A value in the dropdown list changed in the past, and we want to continue to allow legacy vals as valid inputs.
+In the first case, we will alter the validation rule to check against the new correct value, and
+then treat the value currently seen in the worksheet as an allowable legacy value.
+In both cases, we will foribly coerce any instances of legacy values into the correct value.
+This coercion happens whenever we read the value from the upload file, so it will apply to
+validations we perform, as well as values we export in reports.
+   */
+const dropdownCorrections = {
+  'Affordable housing supportive housing or recovery housing': {
+    correctedValue: 'Affordable housing, supportive housing, or recovery housing'
+  },
+  'COVID-19 testing sites and laboratories and acquisition of related equipment': {
+    correctedValue: 'COVID-19 testing sites and laboratories, and acquisition of related equipment'
+  },
+  'Family or child care': {
+    allowableLegacyValues: ['Family or childcare']
+  }
```

Comment:
Do you still need this reverse lookup if we perform formatting before validation?

Reply:
Structuring the lookup this way makes it easier to do a couple things:
1) We can easily verify that the rule is configured to mention a real value in a dropdown somewhere (which I added in `validateDropdownCorrections` in the generate-rules script)
2) It lets us easily add new legacy values if something changes again in the future. E.g. It seems like at some point in the past they changed their capitalization strategy, and then they changed their commas strategy, and then they changed the commas back to the old system. By indexing by the current value and keeping a list of all legacy values, it's easy to add those corrections to the current config.

Reply:
Oh!   You're working around the fact that the latest _input template_ that Joe published still has values that aren't corrected?

Sorry, this took me a long time to grok.

Oof, that's a bit of a pain.

I'd be a lot happier if we only had one format for specifying corrections, but I guess we'd have to sacrifice the validation you wrote to achieve that?

> 1) The value in the current worksheet is incorrect.

If I've got it now, this is referring to the currently committed input template, not the worksheet that you are currently processing.  Is that right?

I'm happy to worry about refactoring later, but (if I have it right) would you mind updating this comment for clarity?

> 1) The value in the currently committed input template is incorrect.

Reply:
yes, that's correct! I'll update the comment now to reference the currently committed input template